### PR TITLE
extend SessionManagerObserver

### PR DESCRIPTION
**Pull request by johnxnguyen**
## Why?
The implementation of multiple account notification sounds (in the UI) needs to know when a `ZMUserSession` has been created or destroyed. We extend the `SessionManagerObserver` with the optional method `sessionManagerDestroyedUserSession(for accountID:)` so that existing observers still conform to the protocol.

## Edit
I've renamed the `SessionManagerDelegate` method `sessionManagerCreated(userSession:)` to `sessionManagerActivated(userSession:)` because this method is only called explicitly when a session is activated, not necessarily when a new user session is created.

Also, the session created and destroyed notifications have been split into two separate observers. The new session notification is posted when brand new user session is created (within `startBackgroundSession(...)`) and the destroyed session notification is posted when a session is torn down or the session is logged out.

### CODE REVIEW

File: Source/SessionManager/SessionManager.swift [hunk 1]
```diff
@@ -781,21 +782,34 @@ extension SessionManager : PreLoginAuthenticationObserver {
...
 @objc public protocol SessionManagerObserver: class {
     func sessionManagerCreated(userSession : ZMUserSession)
+    @objc optional func sessionManagerDestroyedUserSession(for accountId : UUID)
 }
 
 private let sessionManagerObserverNotificationName = Notification.Name(rawValue: "ZMSessionManagerObserverNotification")
 
 extension SessionManager: NotificationContext {
     
+    private var createdKey: String { return "created" }
```
Comment (mikeger) [inline Source/SessionManager/SessionManager.swift]: Can be replaced with `let createdKey = "created"`

File: Source/SessionManager/SessionManager.swift [hunk 2]
```diff
@@ -781,21 +782,34 @@ extension SessionManager : PreLoginAuthenticationObserver {
 // MARK: - Session manager observer
 @objc public protocol SessionManagerObserver: class {
     func sessionManagerCreated(userSession : ZMUserSession)
+    @objc optional func sessionManagerDestroyedUserSession(for accountId : UUID)
 }
 
 private let sessionManagerObserverNotificationName = Notification.Name(rawValue: "ZMSessionManagerObserverNotification")
```
Comment (mikeger) [inline Source/SessionManager/SessionManager.swift]: Should it be two distinctive notifications, one for creation and one for destruction?

**Comment by johnxnguyen** [comment]
@mikeger do you think it's bad to notify for new user sessions when loading a background session?

File: Source/SessionManager/SessionManager.swift (hunk 2 above)
Comment (johnxnguyen) [reply Source/SessionManager/SessionManager.swift]: yeah, it might be neater to have the separation

**Comment by mikeger** [comment]
@johnxnguyen it should be fine, may be we can add a comment that would describe the circumstances when the session can be created and destroyed.

File: Source/SessionManager/SessionManager.swift (hunk 1 above)
Comment (johnxnguyen) [reply Source/SessionManager/SessionManager.swift]: unfortunately we're in an extension here

Comment (mikeger) [reply Source/SessionManager/SessionManager.swift]: Eh, may be it can be static then?

Comment (johnxnguyen) [reply Source/SessionManager/SessionManager.swift]: oh yeah.... 😄

**Comment by johnxnguyen** [comment]
@mikeger I've added some tests 😄

**Comment by mikeger** [comment]
@johnxnguyen can this be merged?

**Comment by johnxnguyen** [comment]
@mikeger yes we're good to go. If you can trigger a build then I can bump components for the UI

**Pull request merged**